src/main/java/com/cflint/plugins/core/ComponentPath.java

```diff
@@ -0,0 +1,108 @@
+package com.cflint.plugins.core;
+
+import org.apache.tools.ant.taskdefs.Java;
+
+import java.awt.*;
+import java.io.File;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.io.IOException;
+
+public class ComponentPath {
+    private static HashMap<String, HashSet<String>> componentsMapCache;
+    private Path rootPath;
+    private static ComponentPath single_instance = null;
+    private String separator;
+    private static HashSet<String> CFknownScriptFunctions;
+
+    public static ComponentPath getInstance(String folder) {
+        if (single_instance == null)
+            single_instance = new ComponentPath(folder);
+
+        return single_instance;
+    }
+
+    private ComponentPath(String folderPath) {
+        //work around to find web root
+        // initialize known sub folders of web root
+        //without this, to find the web root , we have to search for server/web.xml coldfusion configuration file
+        String[] knownRootSubDirs = {"api", "voucherengine", "components", "avisfuelcard", "commandchain", "ioc", "parrot", "paycard", "santam", "voucherpos", "access", "customTags", "common"};
```

Comment:
These should be configurable in an external given file or environment variable.

Reply:
👍